internal/hotfix/retain_line_break.go

```diff
@@ -31,38 +31,42 @@ type paddinger struct {
 	strings.Builder
 }
 
-func (p *paddinger) adjust(size int) {
+func (p *paddinger) adjust(txt string) {
+	var indentSize int
+	for i := 0; i < len(txt) && txt[i] == ' '; i++ { // yaml only allows space to indent.
+		indentSize++
+	}
 	// Grows if the given size is larger than us and always return the max padding.
-	for diff := size - p.Len(); diff > 0; diff-- {
+	for diff := indentSize - p.Len(); diff > 0; diff-- {
 		p.WriteByte(' ')
 	}
 }
 
-func MakeFeatureRetainLineBreak(linebreakStr string, indent int) yamlfmt.Feature {
+func MakeFeatureRetainLineBreak(linebreakStr string) yamlfmt.Feature {
 	return yamlfmt.Feature{
 		Name:         "Retain Line Breaks",
-		BeforeAction: replaceLineBreakFeature(linebreakStr, indent),
+		BeforeAction: replaceLineBreakFeature(linebreakStr),
 		AfterAction:  restoreLineBreakFeature(linebreakStr),
 	}
 }
 
-func replaceLineBreakFeature(newlineStr string, indentSize int) yamlfmt.FeatureFunc {
+func replaceLineBreakFeature(newlineStr string) yamlfmt.FeatureFunc {
 	return func(content []byte) ([]byte, error) {
 		var buf bytes.Buffer
 		reader := bytes.NewReader(content)
 		scanner := bufio.NewScanner(reader)
 		var padding paddinger
 		for scanner.Scan() {
 			txt := scanner.Text()
-			padding.adjust(indentSize)
+			padding.adjust(txt)
 			if strings.TrimSpace(txt) == "" { // line break or empty space line.
 				buf.WriteString(padding.String()) // prepend some padding incase literal multiline strings.
 				buf.WriteString(lineBreakPlaceholder)
 				buf.WriteString(newlineStr)
 				continue
 			}
 			buf.WriteString(txt)
-			buf.WriteByte('\n')
+			buf.WriteString(newlineStr)
```

Comment:
Ah nice catch, I missed that. I'll have to add some `crlf` test cases some time soon.